### ISSUE TITLE
New analyzer: Ensure consistent LINQ style (method vs. query style)

### ISSUE THREAD

**Issue by Greg-Smulko** [dotnet/roslyn#65788]
**Brief description:**

I want to ensure that all LINQ queries across the solution are written in the same syntax - in my case method syntax rather than query syntax.

**Languages applicable:**

Probably C# only? I have no idea. Is VB still a thing? ;P 

**Code example that the analyzer should report:**

E.g., instead of having a mix of:
```cs
var meth = list.Where(x => x.Contains("sth"));

var quer = from l in list
           where l.Contains("sth_else")
           select l;
```

I'd like always to ensure that method syntax is used:
```cs
var meth = list.Where(x => x.Contains("sth"));
var quer = list.Where(x => x.Contains("sth_else"));
```

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
